tuxemon/event/actions/npc_wander.py

```diff
@@ -66,8 +83,16 @@ def move(world: WorldState, npc: NPC) -> None:
             exits = world.get_exits(origin)
             if exits:
                 if origin != coords:
-                    npc.path = [random.choice(exits)]
-                    npc.next_waypoint()
+                    path = random.choice(exits)
+                    if output:
+                        if path in output:
+                            npc.path = [path]
+                            npc.next_waypoint()
+                        else:
+                            return
+                    else:
+                        npc.path = [path]
+                        npc.next_waypoint()
```

Comment:
The imbrication levels of blocks starts to become a bit big... Maybe we could try to lower the count and remove the duplicated code by merging the conditions.

```py
path = random.choice(exits)
if not output or path in output:
    npc.path = [path]
    npc.next_waypoint()
```